Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdaterStrategyResolver.cs

```diff
@@ -45,89 +47,118 @@ public StrategyType GetFallbackStrategy(StrategyType strategyType)
+        public StrategyType Resolve([NotNull] AppUpdaterContext context)
         {
-            Checks.ArgumentNotNull(context, "context");
-
-            DebugLogger.Log("Resolving best strategy for updating...");
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
 
-            if (context.App.IsInstalled())
+            try
             {
-                int installedVersionId = context.App.GetInstalledVersionId();
-                int latestVersionId = context.App.GetLatestVersionId();
+                _logger.LogDebug("Resolving best strategy for updating...");
 
-                if (installedVersionId == latestVersionId)
+                if (context.App.IsInstalled())
                 {
-                    DebugLogger.Log("Installed version is the same as the latest version. Using empty strategy.");
+                    int installedVersionId = context.App.GetInstalledVersionId();
+                    _logger.LogTrace("installedVersionId = " + installedVersionId);
 
-                    return StrategyType.Empty;
-                }
+                    int latestVersionId = context.App.GetLatestVersionId();
+                    _logger.LogTrace("latestVersionId = " + latestVersionId);
+
+                    if (installedVersionId == latestVersionId)
+                    {
+                        _logger.LogDebug("Installed version is the same as the latest version. Using empty strategy.");
+
+                        return StrategyType.Empty;
+                    }
+
+                    if (installedVersionId < latestVersionId)
+                    {
+                        _logger.LogDebug("Installed verison is older than latest version...");
 
-                if (installedVersionId < latestVersionId)
-                {
 #if PATCHKIT_DONT_USE_DIFF_UPDATES
-                    DebugLogger.Log(
+                    _logger.LogDebug(
                         "Installed version is older than the latest version. " +
                         "Using content strategy due to define PATCHKIT_DONT_USE_DIFF_UPDATES");
-                    
+
                     return StrategyType.Content;
 #else
-                    DebugLogger.Log(
-                        "Installed version is older than the latest version. Checking whether cost of updating with diff is lower than cost of updating with content...");
-
-                    if (context.Configuration.CheckConsistencyBeforeDiffUpdate)
-                    {
-                        DebugLogger.Log("Checking consitency before allowing diff update...");
+                        int lowestVersionWithDiffId = GetLowestVersionWithDiffId(context);
+                        _logger.LogTrace("lowestVersionWithDiffId = " + lowestVersionWithDiffId);
 
-                        if (!IsVersionIntegral(context))
+                        if (installedVersionId + 1 < lowestVersionWithDiffId)
```

Comment:
I have some issues here:
1. The next log says `"Diffs for updating from installed to latest version are not available. Using content strategy."`, which, while technically true, does not seem exactly equivalent to what this condition is checking. Shouldn't the log look more like `"The installed version doesn't support diff upadtes. Using content strategy."`?

2. It took me a few seconds to understand the `+ 1` bit. A method along the lines of `DoesVersionSupportDiffUpdates(installedVersionId)` would be more readable.

Reply:
Ad. 2